sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -899,9 +899,13 @@ export class ServiceBusMessageImpl implements ReceivedMessageWithLock {
     context: ClientEntityContext,
     msg: AmqpMessage,
     delivery: Delivery,
-    shouldReorderLockToken: boolean
+    shouldReorderLockToken: boolean,
+    receiveMode: ReceiveMode
   ) {
     Object.assign(this, fromAmqpMessage(msg, delivery, shouldReorderLockToken));
+    if (receiveMode === ReceiveMode.receiveAndDelete) {
```

Comment:
Let's just pop a comment in here saying this is a workaround (because eventually we can get rid of this, right?)

Reply:
Added a comment in https://github.com/Azure/azure-sdk-for-js/pull/10396/commits/6fec19914d6db0e228f851e3f3e3156536b4988f